internal/observability/metrics.go

```diff
@@ -0,0 +1,57 @@
+package observability
+
+import (
+	"context"
+	"sync"
+)
+
+type contextKey string
+
+const keyMetricsRecorder contextKey = "metricsRecorder"
+
+type (
+	Metric interface {
+		IsMetric()
+	}
```

Comment:
if this is going in a generic `observability` package, it should probably have a more specific name (like `SessionMetric` or `SlabTransferMetric` or something)

Reply:
updated